### PR TITLE
Patterns: remove "Manage all parts" page & link

### PR DESCRIPTION
**Pull request by oandregal**
Part https://github.com/WordPress/gutenberg/issues/59659 and https://github.com/WordPress/gutenberg/issues/55083

## What?

- Removes the link to "Manage all template parts" in patterns.
- The `/wp_template_part/all` URL is still accessible to hybrid themes (classic themes that declare support for `block-template-part`).
- The `/patterns` URL is still accessible unofficially to classic themes (with or without support for `block-template-part`).

| | Before | After |
| --- | --- | --- |
| **Block themes**. The "Manage all parts" link is no longer accessible. | Captura de ecrã 2024-04-22, às 12 47 32 | Captura de ecrã 2024-04-22, às 12 55 08 |
| **Hybrid themes**. They still can access the `/wp_template_part/all` URL, as before | Captura de ecrã 2024-04-22, às 12 47 49 | Captura de ecrã 2024-04-22, às 13 10 32 |
| **Hybrid themes**. They still can access the unofficially supported `/patterns` URL, as before | Captura de ecrã 2024-04-22, às 12 48 15 | Captura de ecrã 2024-04-22, às 13 10 39 |
| **Classic themes**. They still can access the unofficially supported `/patterns` URL, as before | Captura de ecrã 2024-04-22, às 12 48 33 | Captura de ecrã 2024-04-22, às 13 00 00 |

## Why?

- That's the target design we want, as per https://github.com/WordPress/gutenberg/issues/59659
- Over small PRs, all functionality have been migrated to the existing Parts pages in Patterns, so this is just removing the link and the code.

## How?

> The `/wp_template_part/all` URL still exists, because hybrid themes require it.

This PR updates the components that are used for that route. It now loads the code from Patterns (sidebar & content components). The reason for this is that in a future step hybrid themes will also have access to the Patterns from the site editor, so we'll load the same components. 

Changes:

- Router: `packages/edit-site/src/components/layout/router.js`
  - Updates the components used. From `PageTemplatesTemplateParts` to `PagePatterns` and from `SidebarNavigationScreenTemplatesBrowse` to `SidebarNavigationScreenPatterns`.
- Patterns
  - Sidebar: `packages/edit-site/src/components/sidebar-navigation-screen-patterns/index.js`. Configures the component depending on whether we're loading the `/wp_template_part/all` URL (display only Template parts) or `/patterns` (display everything, Patterns & Parts).
  - Content: `packages/edit-site/src/components/page-patterns/index.js`. Configures the component depending on whether we're loading the `/wp_template_part/all` URL (display only Template parts) or `/patterns` (display everything, Patterns & Parts).
- Pattern (details view)
  - Sidebar: `packages/edit-site/src/components/sidebar-navigation-screen-pattern/index.js`. Updates the logic so that is goes back to `/wp_template_part/all` if the theme is not a block theme.

> Remove no longer necessary code.

The "Manage template parts" screen used different code to render the Template parts than the "Patterns" screen uses. This PR removes all of that.

Changes:

- Sidebar (`packages/edit-site/src/components/sidebar-navigation-screen-templates-browse/`):
  - `index.js`: removes all the code related to parts.
- Content (`packages/edit-site/src/components/page-templates/`):
  - Renames the directory from `packages/edit-site/src/components/page-templates-template-parts/` to `packages/edit-site/src/components/page-templates/`.
  - Renames the `PageTemplatesTemplateParts` component to `PageTemplates`.
  - Removes `packages/edit-site/src/components/page-templates-template-parts/add-new-template-part.js`, as we already have the same code in the Patterns screen.
  - `index.js`: removes all the code that was used for template parts.

## Testing Instructions

Using a block theme:

- Visit the Patterns page and verify that the link is not there.
- Verify the link back from details takes you back to the patterns page, whether you are editing a Part or a Pattern.

Using a classic theme (for example, visit `localhost:8889` and select TwentyTwentyOne):

- Visit the link `http://localhost:8889/wp-admin/site-editor.php?path=%2Fpatterns` and verify that it works as described above. It's an unofficial link, but people expect it to work.

Using a hybrid theme (for example, visit `localhost:8889` and select EmptyHybrid):

- Go to Appareance > Template parts and verify the site-editor Template Parts page is loaded and works as described above.

- Visit the link `http://localhost:8889/wp-admin/site-editor.php?path=%2Fpatterns` and verify that it works as described above. It's an unofficial link, but people expect it to work.

### CODE REVIEW

**Comment by jameskoster** [comment]
There will be a couple of pieces missing if we move forward with this...

The "Manage all template parts" view supports the List layout, whereas other pattern views do not. We should decide whether it should be possible to view patterns in List layout and apply that across. Probably fine to do that separately if you prefer.

In "Manage all template parts" there are dedicated views to filter by author. This filter is missing entirely in the pattern category views. Would it be possible to add it as a primary filter?

**Comment by oandregal** [comment]
So, there's a question here about how to do with classic themes that declare [theme support](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-support/#block-based-template-parts) for `block-template-parts` (hybrid themes) when we remove the '/wp_template_part/all` page.

This is how they work in 6.5:

- Appareance > Patterns: navigates to wp-admin patterns.
- Appareance > Template Parts: navigates to site editor "Manage all Parts".

https://github.com/WordPress/gutenberg/assets/583546/56e33afe-5b7a-4998-8440-7c8937f7e1b4

This is how they work in this PR:

- Appareance > Patterns: navigates to wp-admin patterns.
- Appareance > Template parts: load site-editor Patterns page, but only has access to Template Parts.

https://github.com/WordPress/gutenberg/assets/583546/94b976bd-6e7c-4614-bafa-24353b4f857d

For this PR, the current state could be enough. However, it raised a few questions for me:

1. Why don't we have an option in the sidebar to view all Parts at once (but we do allow to load the URL).
2. Why don't we load the site-editor Patterns screen for classic themes as well?
3. If we are merging Patterns&Parts in a single screen in the editor for block themes, why don't we do the same for hybrid themes? This is: remove the "Template parts link" in Appareance and leave only the "Patterns" link that would point to the site editor Patterns page (that would display always patterns, but only parts if the theme supports it).

cc @SaxonF @jameskoster @youknowriad 
Also @aaronrobertshaw @kevin940726 @talldan @glendaviesnz as I saw [you](https://github.com/WordPress/gutenberg/pull/54422) having related conversations.

**Comment by oandregal** [comment]
> The "Manage all template parts" view supports the List layout, whereas other pattern views do not. We should decide whether it should be possible to view patterns in List layout and apply that across. Probably fine to do that separately if you prefer.

Yeah, I mentioned this in [the other PR](https://github.com/WordPress/gutenberg/pull/60359#issuecomment-2047031585). I'm happy to implement this as a follow-up if we want, but shouldn't be a blocker IMO.

> In "Manage all template parts" there are dedicated views to filter by author. This filter is missing entirely in the pattern category views. Would it be possible to add it as a primary filter?

mmm, how so? I've added it in a previous step at https://github.com/WordPress/gutenberg/pull/60372

https://github.com/WordPress/gutenberg/assets/583546/fbae6cc4-05af-465b-ac01-d7b83f53a723

**Comment by jameskoster** [comment]
> mmm, how so? I've added it in a previous step at https://github.com/WordPress/gutenberg/pull/60372

My build must not have compiled or something; I wasn't seeing it before, but I do see it now :) 

> Why don't we have an option in the sidebar to view all Parts at once (but we do allow to load the URL)

Since the idea is to consolidate the pattern/template part concepts I think ideally the "All patterns" view will include template parts down the road. Is that technically feasible?

> If we are merging Patterns&Parts in a single screen in the editor for block themes, why don't we do the same for hybrid themes? This is: remove the "Template parts link" in Appareance and leave only the "Patterns" link that would point to the site editor Patterns page (that would display always patterns, but only parts if the theme supports it).

That's a very good question. Considering my previous point, I think your suggestion makes sense. cc @t-hamano on this since you've worked on the classic theme UX.

**Comment by t-hamano** [comment]
> Why don't we load the site-editor Patterns screen for classic themes as well?

This is a planned feature and is being tracked in #52150. For future roadmap, please see comments [here](https://github.com/WordPress/gutenberg/issues/52150#issuecomment-1743235386).

In fact, even in the classic theme, access to the site editor's pattern page (`/wp-admin/site-editor.php?path=%2Fpatterns`) is [internally allowed](https://github.com/WordPress/wordpress-develop/blob/6366755a315b80604740af329eefa8a1a790ff7b/src/wp-admin/site-editor.php#L26) as of WP6.5. However, the Appearance > Patterns menu currently links to the list of pattern posts (`/wp-admin/edit.php?post_type=wp_block`). This is because the Gutenberg plugin needs to support lower versions of WordPress core. Once the Gutenberg plugin supports core version 6.5, we should be able to link the Appearance > Patterns menu to the Patterns page in the site editor.

> If we are merging Patterns&Parts in a single screen in the editor for block themes, why don't we do the same for hybrid themes? This is: remove the "Template parts link" in Appareance and leave only the "Patterns" link that would point to the site editor Patterns page (that would display always patterns, but only parts if the theme supports it).

This should also be possible. However, as mentioned earlier, we should also consider scenarios where the Gutenberg plugin is used on lower WP versions.

**Comment by aaronrobertshaw** [comment]
Thanks for the ping 👍 

I think @t-hamano sums up the current situation well. We should be able to improve some of these things but as I understand it, we'll need to ensure Gutenberg's minimum supported version is 6.5 before the consolidation can completely happen.

**Comment by SaxonF** [comment]
> Why don't we have an option in the sidebar to view all Parts at once (but we do allow to load the URL).

Agree with @jameskoster that end goal is to simplify and treat template parts as synced patterns that belong to special categories (e.g header). In the short term though, if we need to fill a functionality gap, could we add an "all" category under template parts?

Separately, as part of this work, could we move template parts above patterns to give them a little more prominence?

**Comment by oandregal** [comment]
https://github.com/WordPress/gutenberg/pull/60775 adds a "All template parts" section in the sidebar as a previous step so it's available and can be used in this PR for hybrid themes. I'll rebase this when 60775 lands.

**Comment by oandregal** [comment]
@t-hamano @aaronrobertshaw thanks so much for the context.

This PR removes the existing "Manage all template parts" link/page, but it also redirects `/wp_template_part/all` to the Patterns page, so hybrid themes would still have access to template parts — just using the `/patterns` path.

My understanding is that this PR makes progress towards using the site-editor patterns screen for hybrid themes: when that time comes, we'd just need to make everything (patterns&parts) visible.

I'll rebase this PR when https://github.com/WordPress/gutenberg/pull/60775 lands, everything will be a bit clearer then.

**Comment by oandregal** [comment]
> Separately, as part of this work, could we move template parts above patterns to give them a little more prominence?

This is done at https://github.com/WordPress/gutenberg/pull/60359 which I'll update when both https://github.com/WordPress/gutenberg/pull/60775 and this PR land, so we have a better view of everything.

**Comment by t-hamano** [comment]
> In fact, even in the classic theme, access to the site editor's pattern page (`/wp-admin/site-editor.php?path=%2Fpatterns`) is [internally allowed](https://github.com/WordPress/wordpress-develop/blob/6366755a315b80604740af329eefa8a1a790ff7b/src/wp-admin/site-editor.php#L26) as of WP6.5. However, the Appearance > Patterns menu currently links to the list of pattern posts (`/wp-admin/edit.php?post_type=wp_block`). This is because the Gutenberg plugin needs to support lower versions of WordPress core. Once the Gutenberg plugin supports core version 6.5, we should be able to link the Appearance > Patterns menu to the Patterns page in the site editor.

In this regard, by adding a backward compatibility hook to the Gutenberg plugin, we might be able to immediately publish the new Patterns page to all classic themes without having to wait until Gutenberg supports core version 6.5. See [this comment](https://github.com/WordPress/gutenberg/issues/52150#issuecomment-2058160931) for details.

I plan to tackle this challenge.

**Comment by oandregal** [comment]
> > Separately, as part of this work, could we move template parts above patterns to give them a little more prominence?
>
> This is done at https://github.com/WordPress/gutenberg/pull/60359 which I'll update when both https://github.com/WordPress/gutenberg/pull/60775 and this PR land, so we have a better view of everything.

Actually! https://github.com/WordPress/gutenberg/pull/60775 has been merged, so I think it's best if the next step to land is https://github.com/WordPress/gutenberg/pull/60359 This PR is bigger and will benefit from having it merged.

**Comment by oandregal** [comment]
Hi all, this is ready for review.

I've provided some orientation for review in the "How" section of the PR. I wish I could have prepared neat commits as past PRs, though it was too time-consuming for this one given all the rebases and merge conflicts from related work.

@t-hamano @aaronrobertshaw @glendaviesnz @talldan It'd be beneficial if you could test the flow for hybrid themes and report back. In terms of code, I believe this PR will be helpful for hybrid themes to access the site-editor patterns screen in the future: we'd just have to remove the checks in the Pattern's sidebar & content components for the screen to show both Patterns & Template Parts when the time comes.

File: packages/edit-site/src/components/sidebar-navigation-screen-templates-browse/index.js
Comment (mcsf) [inline packages/edit-site/src/components/sidebar-navigation-screen-templates-browse/index.js]: Minor suggestion: since there's no more branching based on postType, why not inline the values? `title={ __( 'Manage templates' ) }`, etc.

**Comment by aaronrobertshaw** [comment]
@oandregal I'm still getting up to speed with all the changes here but it might be of interest that a [backwards compatiblity patch](https://github.com/WordPress/gutenberg/pull/60804) was merged yesterday allowing all themes access to the Patterns page in the site editor. (See https://github.com/WordPress/gutenberg/pull/60804).

**Review by aaronrobertshaw — COMMENTED** [review]
Appreciate all the hard work in wrangling these updates @oandregal 👍 

I've taken this PR for a spin and it mostly works as advertised. 

✅ Manage all parts link has been removed
✅ After viewing/editing a template part, the back button goes back to the Patterns page
✅ Visiting Appearance > Template Parts takes hybrid theme to the Patterns page
✅ Patterns page accessed via Template Parts link for hybrid theme is labelled accordingly

It appears some further work is needed to use the Patterns page for displaying only Template Parts when accessed via Appearance > Template Parts for a hybrid theme. 

The actions available in the sidebar navigation screen allow creating & importing patterns. Is that desired or expected?

| Trunk | This PR |
|---|---|
| Screenshot 2024-04-19 at 1 03 06 PM | Screenshot 2024-04-19 at 1 02 13 PM |



> My understanding is that this PR makes progress towards using the site-editor patterns screen for hybrid themes: when that time comes, we'd just need to make everything (patterns&parts) visible.

My understanding of the discussion on this PR so far is that the end goal is:
- To have only the Patterns page
- The Patterns page covers both patterns and template parts
- For all themes to be able to access the Patterns page in the site editor

To that end, now https://github.com/WordPress/gutenberg/pull/60804 has merged providing the backwards compatibility to allow access to the patterns page for all themes, we can update this PR to only have the combined page showing both? Am I missing something?

**Comment by t-hamano** [comment]
> My understanding of the discussion on this PR so far is that the end goal is:
> - To have only the Patterns page
> - The Patterns page covers both patterns and template parts
> - For all themes to be able to access the Patterns page in the site editor

I also believe that this is the end goal.

> Visiting Appearance > Template Parts takes hybrid theme to the Patterns page

In my testing, Appearance > Template Parts displays the All Template Parts page instead of the Patterns page 🤔

https://github.com/WordPress/gutenberg/assets/54422211/0fb22835-92d4-4e58-a814-9b0472b174d5

**Comment by aaronrobertshaw** [comment]
> In my testing, Appearance > Template Parts displays the All Template Parts page instead of the Patterns page 🤔

@t-hamano, apologies, what I meant in that comment was that the page is rendering the Patterns page components and is filtered to template parts only. That's why you'll note the Pattern related actions that aren't there on trunk's manage template parts page.

André notes this expectation in the PR description:

> This PR updates the components that are used for that route. It now loads the code from Patterns (sidebar & content components). 

So it's still the template parts URL but the page's underlying components are those from the Patterns page.

**Comment by oandregal** [comment]
> To that end, now https://github.com/WordPress/gutenberg/pull/60804 has merged providing the backwards compatibility to allow access to the patterns page for all themes, we can update this PR to only have the combined page showing both? Am I missing something?

That requires looking at a few other things, I believe: updating all the places that point to the wp-admin Patterns screen and redirect them; consider why we'd have two links under Appearance (patterns&parts) if they point to the same screen; etc. It merits to be looked at separately.

Additionally, @t-hamano has also expressed interest in working on that area ("[Next steps](https://github.com/WordPress/gutenberg/pull/60804)"), so it'd be good in we can organize the work in a way that we can have more done by parallelizing tasks — which we can do if we land this PR as it is.

**Comment by oandregal** [comment]
> The actions available in the sidebar navigation screen allow creating & importing patterns. Is that desired or expected?

Missed that. Thanks for reporting! It's now fixed https://github.com/WordPress/gutenberg/pull/60689/commits/2ea07e2491ac2b6bc89e60009d87fc26ed53dd68

**Review by t-hamano — COMMENTED** [review]
This PR is working as expected overall, but there is something strange when the hybrid theme accesses `http://localhost:8889/wp-admin/site-editor.php?path=%2Fpatterns`.

The sidebar shows Template Parts, while the canvas shows Patterns:

image

If we have WordPress 6.5 or higher, or if we have the latest Gutenberg enabled, all classic themes have internal access to the new Patterns page, so we always need to show the Patterns page navigation.

This is a screenshot of the current trunk:

image

**Comment by t-hamano** [comment]
> Additionally, @t-hamano has also expressed interest in working on that area ("[Next steps](https://github.com/WordPress/gutenberg/pull/60804)"), so it'd be good in we can organize the work in a way that we can have more done by parallelizing tasks 

I will be happy to work with you. I think that in addition to the tasks listed in "Next Steps", the Apperance > Template Parts submenu is removed in hybrid themes. Ultimately, all classic themes will only show Apperance > Patterns, which links to the new Patterns page in the Site Editor. The theme template parts bundled with a hybrid theme can be accessed from its Patterns page. Is my understanding correct?

**Comment by oandregal** [comment]
So, there's a few scenarios to cover: block themes, hybrid themes (two separate paths), and classic themes. I've pushed the changes to make them all work and updated the PR description.

This is the before/after:

| | Before | After |
| --- | --- | --- |
| **Block themes**. The "Manage all parts" link is no longer accessible. | Captura de ecrã 2024-04-22, às 12 47 32 | Captura de ecrã 2024-04-22, às 12 55 08 |
| **Hybrid themes**. They still can access the `/wp_template_part/all` URL, as before | Captura de ecrã 2024-04-22, às 12 47 49 | Captura de ecrã 2024-04-22, às 13 10 32 |
| **Hybrid themes**. They still can access the unofficially supported `/patterns` URL, as before | Captura de ecrã 2024-04-22, às 12 48 15 | Captura de ecrã 2024-04-22, às 13 10 39 |
| **Classic themes**. They still can access the unofficially supported `/patterns` URL, as before | Captura de ecrã 2024-04-22, às 12 48 33 | Captura de ecrã 2024-04-22, às 13 00 00 |

**Comment by oandregal** [comment]
I've also tested the backflows from the editor and this is how it works:

1. Block themes ✅ 

https://github.com/WordPress/gutenberg/assets/583546/61dff745-2bb6-44c9-b327-53be40237fc6

2. Classic themes (unofficial patterns path) ✅ 

https://github.com/WordPress/gutenberg/assets/583546/66accdf0-21de-420c-b039-78b9c9c88173

3. Hybrid themes (template parts path) ✅ 

https://github.com/WordPress/gutenberg/assets/583546/fdfba007-c1e8-4bcf-8a68-f3f8e79400ce

4. Hybrid themes (unofficial patterns path). Note how the backpath navigates the user to the Template Parts screen instead of the Patterns screen where they came from. We don't have a way to know _where_ it came from, so it cannot be fixed here. The fix is actually the next step in this consolidation: making the `/patterns` path official and remove the template parts path. Given this is an "unofficial" page and that we plan to consolidate in this cycle, I think this is ok for now.

https://github.com/WordPress/gutenberg/assets/583546/d7f249de-414c-4214-9ca0-68b1f4de962f

**Comment by oandregal** [comment]
Unless there's more use cases to support, I think this is working as expected. Is there anything else left or can this be merged?

File: packages/edit-site/src/components/sidebar-navigation-screen-pattern/index.js
Comment (mcsf) [inline packages/edit-site/src/components/sidebar-navigation-screen-pattern/index.js:35]: We should document `supportsTemplatePartsMode` and/or state the intention of this `const`.

**Review by mcsf — APPROVED** [review]
LGTM, but you know this better than me. :)

File: packages/edit-site/src/components/sidebar-navigation-screen-pattern/index.js
Comment (oandregal) [reply packages/edit-site/src/components/sidebar-navigation-screen-pattern/index.js:35]: [2dbe1c5](https://github.com/WordPress/gutenberg/pull/60689/commits/2dbe1c5d15ad27deba110c825a3c1852321c440b) should help.

File: packages/edit-site/src/components/sidebar-navigation-screen-templates-browse/index.js
Comment (oandregal) [reply packages/edit-site/src/components/sidebar-navigation-screen-templates-browse/index.js]: Addressed at [5192d86](https://github.com/WordPress/gutenberg/pull/60689/commits/5192d86d60a1ec1f130637b5b21d87dcbd0f4c7b)

File: packages/edit-site/src/components/sidebar-navigation-screen-patterns/index.js
Comment (t-hamano) [inline packages/edit-site/src/components/sidebar-navigation-screen-patterns/index.js]:
```suggestion
			actions={
				( isBlockBasedTheme || ! isTemplatePartsPath ) && (
					<AddNewPattern
						canCreateParts={ isBlockBasedTheme }
						canCreatePatterns={ ! isTemplatePartsPath }
					/>
				)
			}
```

How about changing to conditional rendering? Otherwise, the hidden field will be rendered even though there are no controls.

template-parts-no-control

File: packages/edit-site/src/components/page-patterns/header.js
Comment (t-hamano) [inline packages/edit-site/src/components/page-patterns/header.js]: Is there a reason why you removed "all" from the title and the description fallback? I'm curious about the difference when opening "All patterns" and "All template parts" in the sidebar.

"Al template part" is selected:

image


"All patterns" is selected:

image

**Review by t-hamano — APPROVED** [review]
I left only two suggestions, but they all work as expected. I think there are some tasks to tackle after this is merged, do you have any plans?

File: packages/edit-site/src/components/sidebar-navigation-screen-patterns/index.js
Comment (oandregal) [reply packages/edit-site/src/components/sidebar-navigation-screen-patterns/index.js]: Done at [08e8ebe](https://github.com/WordPress/gutenberg/pull/60689/commits/08e8ebeb81985a8512f6861c858516adda5ba04b)

File: packages/edit-site/src/components/page-patterns/header.js
Comment (oandregal) [reply packages/edit-site/src/components/page-patterns/header.js]: I thought about making these text the same for hybrid themes in the template parts screen so before/after is the same. Though I now realize it's best to leave them as they are. Reverted the changes at [18c6a0b](https://github.com/WordPress/gutenberg/pull/60689/commits/18c6a0bb015f98bf9c7ee956eb7274b679e4c9b7)

**Comment by oandregal** [comment]
> I think there are some tasks to tackle after this is merged, do you have any plans?

I'm going to take a few days off this area: I need to prepare a talk I'll give by end of week + looking at improving the `DataViews` component. My main focus is https://github.com/WordPress/gutenberg/issues/59659 Happy to review work in this area.

Do you have a sense of next steps for this? I presume we want to update the Patterns link for classic themes so they load the site-editor Patterns page. Then also remove `/wp_template_part/all` link, perhaps?

**Comment by jameskoster** [comment]
Thanks for all the work here @oandregal.

In terms of pattern / template part alignment I think there are a couple of outstanding details that it'd be good to address for 6.6 if possible:

1. "All patterns" view should include template parts (probably the most challenging part).
2. The separate menus should ideally converge, mockup here: https://github.com/WordPress/gutenberg/issues/57011#issuecomment-1972878319.
3. We need to settle on a more intuitive name for the "General" category.
4. Align or differentiate the "Header" + "Headers" categories.
5. Update the creation flow, mockup here: https://github.com/WordPress/gutenberg/issues/57011#issuecomment-2023173050

**Comment by t-hamano** [comment]
> I presume we want to update the Patterns link for classic themes so they load the site-editor Patterns page. Then also remove `/wp_template_part/all` link, perhaps?

I think so. I will be working on officially publishing the Site Editor Patterns page for all Classic themes in WP6.6. Detailed specifications are summarized in [this comment](https://github.com/WordPress/gutenberg/issues/52150#issuecomment-2072070889).

I believe it does not conflict with the improvements suggested by @jameskoster.

**Comment by oandregal** [comment]
@jameskoster I've added those tasks as part of the tracking issue, so they're not lost https://github.com/WordPress/gutenberg/issues/59659

**Pull request merged**
